cli/ops/runtime_compiler.rs

```diff
@@ -71,16 +76,57 @@ struct TranspileArgs {
   options: Option<String>,
 }
 
+#[derive(Debug, Serialize)]
+struct RuntimeTranspileEmit {
+  source: String,
+  map: Option<String>,
+}
+
 async fn op_transpile(
   state: Rc<RefCell<OpState>>,
   args: Value,
   _data: BufVec,
 ) -> Result<Value, AnyError> {
   super::check_unstable2(&state, "Deno.transpile");
   let args: TranspileArgs = serde_json::from_value(args)?;
-  let cli_state = super::global_state2(&state);
-  let program_state = cli_state.clone();
-  let result =
-    runtime_transpile(program_state, &args.sources, &args.options).await?;
+
+  let mut compiler_options = tsc_config::TsConfig::new(json!({
+    "checkJs": true,
+    "emitDecoratorMetadata": false,
+    "jsx": "react",
+    "jsxFactory": "React.createElement",
+    "jsxFragmentFactory": "React.Fragment",
+    "esModuleInterop": true,
+    "module": "esnext",
```

Comment:
This setting is currently completely ignored. One of our tests uses `amd` as `module`. We need to implement that for other remaining `module` options as well. Reference https://swc.rs/rustdoc/swc_ecma_transforms/modules/amd/struct.Config.html

Reply:
And it turns out `SystemJS` is not yet supported by SWC.

Reply:
As per Discord conversation I removed option to specify output `module` - actually user supplied config is run through the same set of `IGNORED_COMPILER_OPTIONS` and API returns error when there's an ignored option.

Reply:
It probably needs to be a different set, because things like `target` should be changeable, like `target`, `inlineSourceMaps`, `inlineSources`, etc, which we disallow because they change the emit internally to Deno.  There are probablly two different classes of options... the ones for Deno internal emit we ignore, and ones we ignore totally/don't support.  For `--config` it is both sets, but for the the runtime APIs it is just the second set.

Reply:
Sounds good, I'll rework it